extensions/wikia/PortabilityDashboard/PortabilityDashboardModel.class.php

```diff
@@ -8,23 +8,101 @@ class PortabilityDashboardModel {
 
 	private $connection;
 
-	const WIKIS_LIMIT = 500;
+	// DIANA: CHANGE AFTER DEVELOPMENT
+	const WIKIS_LIMIT = 5;
 
 	public function __construct( $db = null ) {
 		$this->connection = $db;
 	}
 
 	/**
-	 * gets model data for portability dashboard
+	 * gets full (non filtered) list of data for portability dashboard
 	 * @return array
 	 */
 	public function getList() {
 		$rowList = $this->getRowList();
+
+		return $this->extendList( $rowList );
+	}
+
+	/**
+	 * For wiki url in form:
+	 * - http://www.yugioh.wikia.com
+	 * - www.yugioh.wikia.com
+	 * - yugioh.wikia.com
+	 * - yugioh
+	 * return portability data
+	 *
+	 * @param $wikiUrl
+	 * @return array
+	 */
+	public function getWikiByUrl( $wikiUrl ) {
+		$wiki = [ ];
+		$wwwPrefix = 'www.';
+		$httpPrefix = 'http://';
+		$wikiacomSuffix = '.wikia.com';
+
+		if ( strpos( $wikiUrl, $wwwPrefix ) === false ) {
+			$wikiUrl = $wwwPrefix . $wikiUrl;
+		}
+
+		if ( strpos( $wikiUrl, $httpPrefix ) === false ) {
+			$wikiUrl = $httpPrefix . $wikiUrl;
+		}
+
+		if ( strpos( $wikiUrl, $wikiacomSuffix ) === false ) {
+			$wikiUrl = $wikiUrl . $wikiacomSuffix;
+		}
+
+		$wikiId = WikiFactory::UrlToID( $wikiUrl );
+
+		if ( $wikiId ) {
+			$dataRow = $this->getWikiById( $wikiId );
+
+			if ( $dataRow ) {
+				$wiki = $this->extendList( $dataRow );
+			}
+		}
+
+		return $wiki;
+	}
+
+	/**
+	 * @param $rowList
+	 * @return array
+	 */
+	private function extendList( $rowList ) {
 		$wikiParamsList = $this->getWikiParamsList( $rowList );
 
 		return $this->extendRowListWithWikiParams( $rowList, $wikiParamsList );
 	}
 
+	private function getWikiById( $id ) {
+		return ( new WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::PORTABILITY_DASHBOARD_TABLE )
+			->WHERE( 'excluded' )->EQUAL_TO( 0 )
+			->AND_( 'wiki_id' )->EQUAL_TO( $id )
+			->ORDER_BY( 'migration_impact' )
+			->DESC()
+			->LIMIT( 1 )
+			->run( $this->readDB(), function ( ResultWrapper $rows ) {
+				$result = [ ];
+				while ( $row = $rows->fetchObject() ) {
+					$result[] = [
+						'wikiId' => $row->wiki_id,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "wikiId" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)